cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -289,15 +359,18 @@ class device_row_comparator {
         ++depth;
       }
 
-      auto const comparator = element_comparator{_check_nulls, lcol, rcol, _null_precedence, depth};
       return cudf::type_dispatcher<dispatch_void_if_nested>(
-        lcol.type(), comparator, lhs_element_index, rhs_element_index);
+        lcol.type(),
+        element_comparator{_check_nulls, lcol, rcol, _null_precedence, depth, _comparator},
+        lhs_element_index,
+        rhs_element_index);
     }
 
    private:
     column_device_view const _lhs;
     column_device_view const _rhs;
     Nullate const _check_nulls;
+    PhysicalElementComparator const _comparator;
     null_order const _null_precedence;
     int const _depth;
```

Comment:
It's important to list members in the same order they are initialized to avoid this type of warning:
https://stackoverflow.com/questions/24285112/why-must-initializer-list-order-match-member-declaration-order

```suggestion
    null_order const _null_precedence;
    int const _depth;
    PhysicalElementComparator const _comparator;
```

Reply:
I also notice that the order of constructor arguments is not aligned with the member/initialization order. `check_nulls` is first in the constructor, but third (after `_lhs, _rhs`) in the member/initialization order. Maybe re-order the member/initialization to match the constructor order. It's not entirely related to this PR's scope but it probably needs to be done and feels too inconsequential to do this in a separate PR. :confused: